test/BinaryParse.cpp

```diff
@@ -446,7 +490,269 @@ TEST_F(BinaryParseTest, ExtendedInstruction) {
+        //    %1 = OpTypeInt 64 0
+        //    %2 = OpConstant %1 <missing>
+        {Concatenate({ExpectedHeaderForBound(3),
+                      {MakeInstruction(SpvOpTypeInt, {1, 64, 0})},
+                      {spvOpcodeMake(5, SpvOpConstant), 1, 2}}),
+         "End of input reached while decoding OpConstant starting at word"
+         " 9: missing possibly multi-word literal number operand at word "
+         "offset 3."},
+        // Detect when we provide only one word from the 64-bit literal,
+        // and again lie about the number of words in the instruction.
+        {Concatenate({ExpectedHeaderForBound(3),
+                      {MakeInstruction(SpvOpTypeInt, {1, 64, 0})},
+                      {spvOpcodeMake(5, SpvOpConstant), 1, 2, 42}}),
+         "End of input reached while decoding OpConstant starting at word"
+         " 9: truncated possibly multi-word literal number operand at word "
+         "offset 3."},
+        // Detect when a required string operand is missing.
+        // Also, lie about the length of the instruction.
+        {Concatenate({ExpectedHeaderForBound(3),
+                      {spvOpcodeMake(3, SpvOpString), 1}}),
+         "End of input reached while decoding OpString starting at word"
+         " 5: missing literal string operand at word offset 2."},
+        // Detect when a required string operand is truncated: it's missing
+        // a null terminator.  Catching the error avoids a buffer overrun.
+        {Concatenate({ExpectedHeaderForBound(3),
+                      {spvOpcodeMake(4, SpvOpString), 1, 0x41414141,
+                       0x41414141}}),
+         "End of input reached while decoding OpString starting at word"
+         " 5: truncated literal string operand at word offset 2."},
+        // Detect when an optional string operand is truncated: it's missing
+        // a null terminator.  Catching the error avoids a buffer overrun.
+        // (It is valid for an optional string operand to be absent.)
+        {Concatenate({ExpectedHeaderForBound(3),
+                      {spvOpcodeMake(6, SpvOpSource),
+                       uint32_t(SpvSourceLanguageOpenCL_C), 210,
+                       1 /* file id */,
+                       /*start of string*/ 0x41414141, 0x41414141}}),
+         "End of input reached while decoding OpSource starting at word"
+         " 5: truncated literal string operand at word offset 4."},
+
+        // (End of input exhaustion test cases.)
+
+        // In this case the instruction word count is too small, where
+        // it would truncate a multi-word operand to OpConstant.
+        {Concatenate({ExpectedHeaderForBound(3),
+                      {MakeInstruction(SpvOpTypeInt, {1, 64, 0})},
+                      {spvOpcodeMake(4, SpvOpConstant), 1, 2, 44, 44}}),
+         "Invalid word count: OpConstant starting at word 9 says it has 4"
+         " words, but found 5 words instead."},
```

Comment:
No. The error is issued before it hits that last 44.  The parser is working from the assumption that the word count (4) is valid.  But the type information says the constant is 64 bits wide.
The 1 is the result type, the 2 is the result id, and there should be two words in the literal. But the word count of 4 says we shouldn't go that far.
I will add a better comment, and add the case where the second 44 is absent.